dpx/src/dpx_pdfobj.rs

```diff
@@ -3347,69 +3334,57 @@ unsafe fn pdf_get_object(
         let mut objstm_num: u32 = (*(*pf).xref_table.offset(obj_num as isize)).field2;
         let mut index: u16 = (*(*pf).xref_table.offset(obj_num as isize)).field3;
         let mut objstm: *mut pdf_obj = 0 as *mut pdf_obj;
-        if objstm_num >= (*pf).num_obj as u32
-            || (*(*pf).xref_table.offset(objstm_num as isize)).typ as i32 != 1i32
-            || {
+        if !(objstm_num >= (*pf).num_obj as u32)
+            && (*(*pf).xref_table.offset(objstm_num as isize)).typ as i32 == 1i32
+            && {
                 objstm = (*(*pf).xref_table.offset(objstm_num as isize)).direct;
-                !(!objstm.is_null() || {
+                (!objstm.is_null() || {
                     objstm = read_objstm(pf, objstm_num);
                     !objstm.is_null()
                 })
             }
         {
-            current_block = 17536737673648832705;
-        } else {
             let mut data = get_objstm_data(&*objstm);
             let fresh25 = data;
             data = data.offset(1);
             let n = *fresh25;
             let fresh26 = data;
             data = data.offset(1);
             let first = *fresh26;
-            if index as i32 >= n || *data.offset((2i32 * index as i32) as isize) as u32 != obj_num {
-                current_block = 17536737673648832705;
-            } else {
+            if !(index as i32 >= n) && *data.offset((2i32 * index as i32) as isize) as u32 == obj_num {
+                assert!((*objstm).is_stream());
+                let objstm_slice = &(*(*((*objstm).data as *mut pdf_stream)).stream);
+
                 let length = pdf_stream_length(objstm);
-                let mut p = (pdf_stream_dataptr(objstm) as *const i8)
-                    .offset(first as isize)
-                    .offset(*data.offset((2i32 * index as i32 + 1i32) as isize) as isize);
-                let q = p.offset(
-                    (if index as i32 == n - 1i32 {
-                        length
-                    } else {
-                        first + *data.offset((2i32 * index as i32 + 3i32) as isize)
-                    }) as isize,
-                );
-                result = parse_pdf_object(&mut p, q, pf);
-                if result.is_null() {
-                    current_block = 17536737673648832705;
-                } else {
-                    current_block = 13472856163611868459;
-                }
-            }
-        }
-        match current_block {
-            13472856163611868459 => {}
-            _ => {
-                warn!("Could not read object from object stream.");
-                return pdf_new_null();
+                let pdfobj_start = first + *data.offset(2*index as isize+1);
+                let pdfobj_end = if index as i32 == n - 1 { length } else { first + *data.offset(2*index as isize+3) };
+
+                let mut pdfobj_slice = &objstm_slice[pdfobj_start as usize..pdfobj_end as usize];
+                result = pdfobj_slice.parse_pdf_object(pf);
```

Comment:
This is the last usage of the wrapper function https://github.com/Mrmaxmeier/tectonic/blob/e6ee43c3f155692e5ee47a3a974e4169f7b47f6d/dpx/src/dpx_pdfparse.rs#L353-L368

Reply:
Oh my bad, there are a couple more.